analysis/import/SCMLogParser/src/main/kotlin/de/maibornwolff/codecharta/importer/scmlogparser/parser/CommitCollector.kt

```diff
@@ -10,70 +11,97 @@ import java.util.stream.Collector
-                .filter { !versionControlledFilesContainsFile(versionControlledFiles, it) }
-                .forEach { addYetUnknownFile(versionControlledFiles, it) }
-    }
+        commit.modifications.forEach {
+            renamesMap
+            nameConflictsMap
 
-    private fun versionControlledFilesContainsFile(
-        versionControlledFiles: List<VersionControlledFile>,
-        filename: String
-    ): Boolean {
-        return findVersionControlledFileByFilename(versionControlledFiles, filename) != null
-    }
+            val trackName = if (it.oldFilename.isNotEmpty()) it.oldFilename else it.currentFilename
+            var possibleConflictName: String = if (nameConflictsMap.containsKey(trackName)) {
+                trackName + "_\\0_" + nameConflictsMap[trackName]
+            } else {
+                trackName
+            }
 
-    private fun findVersionControlledFileByFilename(
-        versionControlledFiles: List<VersionControlledFile>,
-        filename: String
-    ): VersionControlledFile? {
-        return versionControlledFiles.firstOrNull { it.filename == filename }
-    }
+            when (it.type) {
+                Modification.Type.ADD -> {
+                    val file: VersionControlledFile? =
+                        versionControlledFiles[possibleConflictName] //does the file exist?
+                    if (file == null) {
+                        val missingVersionControlledFile =
+                            VersionControlledFile(possibleConflictName, metricsFactory) // if not create a file
+                        versionControlledFiles[possibleConflictName] =
+                            missingVersionControlledFile // and add it to the list
+                        missingVersionControlledFile.registerCommit(commit, it)
+                    } else {
+                        versionControlledFiles[possibleConflictName]?.registerCommit(commit, it)
+                    }
+                }
+                Modification.Type.DELETE -> {
+                    val filename = renamesMap[possibleConflictName]
+                    versionControlledFiles.remove(if (filename == null) possibleConflictName else filename) //remove a deleted file from list
+                    renamesMap.remove(possibleConflictName) // and remove its references in renames
+                }
+                Modification.Type.RENAME -> {
 
-    private fun addYetUnknownFile(
-        versionControlledFiles: MutableList<VersionControlledFile>,
-        filenameOfYetUnversionedFile: String
-    ): Boolean {
-        val missingVersionControlledFile = VersionControlledFile(filenameOfYetUnversionedFile, metricsFactory)
-        return versionControlledFiles.add(missingVersionControlledFile)
-    }
+                    val renameName: String? = renamesMap[possibleConflictName] //current
+                    val tmp: String? =
+                        if (renameName == null) possibleConflictName else renameName //do we have an entry?
```

Comment:
We probably have to check move these two variables to the top before `when`. That way `add` and `modify` are also going to use the renamed files.